src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala

```diff
@@ -258,7 +258,7 @@ class PodsResource @Inject() (
     // don't need to authorize as taskKiller will do so.
     withValid(id.toRootPath) { id =>
       withValid(instanceId) { instanceId =>
-        val instances = result(taskKiller.kill(id, _.filter(_.instanceId == Instance.Id(instanceId))))
+        val instances = result(taskKiller.kill(id, _.filter(_.instanceId == Instance.Id.fromIdString(instanceId))))
```

Comment:
nit: it would be nice to evaluate `Instance.Id.fromIdString(instanceId)` only once.

Reply:
✅

Reply:
Wow, this was actually hiding a bug in the tests.